zitadel/v2/smtp_config/resource.go

```diff
@@ -0,0 +1,49 @@
+package smtp_config
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func GetResource() *schema.Resource {
+	return &schema.Resource{
+		Description: "Resource representing the SMTP configuration of an instance.",
+		Schema: map[string]*schema.Schema{
+			senderAddressVar: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Address used to send emails.",
+			},
+			senderNameVar: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Sender name used to send emails.",
+			},
+			tlsVar: {
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: "TLS used to communicate with your SMTP server.",
+			},
+			hostVar: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Host address to your SMTP server.",
+			},
+			userVar: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "User used to communicate with your SMTP server.",
+			},
+			passwordVar: {
+				Type:        schema.TypeString,
+				Required:    true,
```

Comment:
is that really required?

Reply:
In ZITADEL it's no required, but as you should be able to use an empty string it's easier to handle like that

Reply:
So, your idea is, that users should always specify all fields, without having the concept of defaults at all?

Reply:
No? as there is no default definition for the password at all, I rather have a fully defined resource in terraform, as with only 5 attributes it has no impact. As SMTP settings are such a key resource.